### ISSUE TITLE
How can I make three windows vertically with equal spacing?

### ISSUE THREAD

**Issue by IndianBoy42** [pop-os/shell#1177]
I can of course do this manually, although it is a bit annoying and counterintuitive to do so with just the keyboard. But can I distribute the space evenly automatically? maybe even have the splits be like this by default

**Comment by mmstick** [comment]
Not automatically

**Comment by pesader** [comment]
I find myself wanting this all the time when using a 21:9 monitor. Since the first split gets you two 10.5:9 windows, the second split occurs in the same direction as the first leaving you with two thin (and not very useful) 5.25:9 windows.
![image](https://user-images.githubusercontent.com/65264536/133470166-abb4f0c5-b721-4553-ad68-d9df860680cc.png)

It would be awesome if pop shell automatically sized windows equally in the direction they were **twice** splitted at, like so:
![image](https://user-images.githubusercontent.com/65264536/133470192-60c9d1d6-0389-4c7b-9e1a-f74072c8e554.png)

**Comment by IndianBoy42** [comment]
Even more generally after getting my new ultrawide monitor, I've found myself wanting to script and more precisely design window layouts.

One pattern i like (apart from 3 equals) is a fairly wide central window 16:9 or 3:2 with my primary task, with 2 flanking auxiliary windows with side info/music player etc. Currently AFAICT windows are organized as binary trees, but i think ultrawides show they should really be arbitrary trees (or at least be able to be ternary. 

Another one is inspired by https://github.com/beauwilliams/focus.nvim. which splits neovim into golden ratio sized windows, and also increases the size of the focused window as it changes

Is there any plan for allowing scripting of the pop shell TWM in a way that i3, etc have? Or maybe just implementing the mentioned layout strategies

**Comment by pesader** [comment]
I tried using [lwsm](https://github.com/johannesjo/linux-window-session-manager) ("linux window session manager") to save and restore layouts, but it only worked partially. As you can see, it correctly restores this bizarre layout I made just for demonstration IF TILING IS TURNED OFF:

https://user-images.githubusercontent.com/65264536/133474963-c318eb89-c9a0-4d09-86e1-4cc55ab88b32.mp4

However, as soon as I turn tiling on again (at 00:13 timestamp of the video, it is a bit hard to spot since all windows look the same) it dismantles the restored layout.

https://user-images.githubusercontent.com/65264536/133475182-019302b0-7993-4e6b-84bb-e76e97ff0d26.mp4

Still, considering how well `lwsm` restored the layout maybe it is worth investigating some way to make it play nicely with pop shell (perhaps even integrated session management, like Material Shell does).

**Comment by IndianBoy42** [comment]
Are there actually any plans or will to implement saving/restoring layouts, custom sizing, scripting, etc that would allow us to do better than splitting in half every time then manually finagling the layout to how we want (never getting exactly one thirds because we can't)

**Comment by wojpawlik** [comment]
https://material-shell.com/ can already split into thirds, and auto-save and restore layout.

**Comment by Gu7z** [comment]
@mmstick Any plans to make this work automatically? 3 Columns with same size :eyes:

**Comment by Gu7z** [comment]
@mmstick, any news?

**Comment by jtheoof** [comment]
More generally speaking, the way other tiling window managers such as i3 or sway do it is that they have the concept of "direction" that user sets. It's "vertical" or "horizontal". And the window manager adjusts all windows when a new one is created depending on the direction. It would nice if this was implemented in Pop shell as well.

**Comment by paulust2002** [comment]
Agree with the vertical part for sure as generally unusable for most apps if not equal. 

Also would love when reverting from say three vertical equal slices and closing or minimizing an app it automatically resizes equally to two rather than one staying two thirds of screen.

Great work though, switched from Forge as issues with it excluding browsers from tiling in settings every boot.

**Comment by leviport** [comment]
Setting up a triple split in COSMIC DE is pretty easy:

https://github.com/user-attachments/assets/0bbe4758-ba47-4142-9142-a008c56f9d44

**Comment by IndianBoy42** [comment]
Yeah this can be closed, pop shell is not under active development and the interactive triple split in cosmic DE is very nice
